### PR TITLE
[project-base] SSP-2279 - extended datafixtures by subcategories in the menu

### PR DESCRIPTION
**Pull request by AmpMVn**
#### Description, the reason for the PR
Added subcategory to menu in datafixtures

#### Check the appropriate checkboxes below, please

- [ ] [BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/) 
- [ ] New feature 
- [x] I have read and signed [License Agreement for contributions](https://www.shopsys.com/license-agreement)

----
:globe_with_meridians: Live Preview:
  - https://mvn-ssp-2279-fixtures-add-subcategories-to-navigation.odin.shopsys.cloud
  - https://cz.mvn-ssp-2279-fixtures-add-subcategories-to-navigation.odin.shopsys.cloud

### CODE REVIEW

File: project-base/app/src/DataFixtures/Demo/NavigationItemDataFixture.php [hunk 1]
```diff
@@ -106,20 +160,20 @@ public function getDependencies(): array
 
     /**
      * @param \Shopsys\FrameworkBundle\Model\Navigation\NavigationItemData $navigationItemData
+     * @param array $categoriesByColumn
```
Comment (vitek-rostislav) [inline project-base/app/src/DataFixtures/Demo/NavigationItemDataFixture.php]:
```suggestion
     * @param array<int, string[]> $categoriesByColumn
```

File: project-base/app/src/DataFixtures/Demo/NavigationItemDataFixture.php [hunk 2]
```diff
@@ -106,20 +160,20 @@ public function getDependencies(): array
...
-        $navigationItemData->categoriesByColumnNumber[3] = [
-            $this->getCategoryReference(CategoryDataFixture::CATEGORY_FOOD),
-        ];
+    private function addCategoriesToNavigationItem(
+        NavigationItemData $navigationItemData,
+        array $categoriesByColumn,
+    ): void {
+        foreach ($categoriesByColumn as $columnNumber => $categories) {
+            $navigationItemData->categoriesByColumnNumber[$columnNumber] = array_map(
+                function ($category) {
```
Comment (vitek-rostislav) [inline project-base/app/src/DataFixtures/Demo/NavigationItemDataFixture.php]:
```suggestion
                function (string $category) {
```

File: project-base/app/src/DataFixtures/Demo/NavigationItemDataFixture.php [hunk 3]
```diff
@@ -106,20 +160,20 @@ public function getDependencies(): array
...
-        ];
-        $navigationItemData->categoriesByColumnNumber[2] = [
-            $this->getCategoryReference(CategoryDataFixture::CATEGORY_GARDEN_TOOLS),
-        ];
-        $navigationItemData->categoriesByColumnNumber[3] = [
-            $this->getCategoryReference(CategoryDataFixture::CATEGORY_FOOD),
-        ];
+    private function addCategoriesToNavigationItem(
+        NavigationItemData $navigationItemData,
+        array $categoriesByColumn,
```
Comment (vitek-rostislav) [inline project-base/app/src/DataFixtures/Demo/NavigationItemDataFixture.php]: I would rather name this parameter `$categoryReferenceNamesByColumn`

File: project-base/app/src/DataFixtures/Demo/NavigationItemDataFixture.php [hunk 4]
```diff
@@ -106,20 +160,20 @@ public function getDependencies(): array
...
+    ): void {
+        foreach ($categoriesByColumn as $columnNumber => $categories) {
```
Comment (vitek-rostislav) [inline project-base/app/src/DataFixtures/Demo/NavigationItemDataFixture.php]: ...and this `$categoryReferenceNames`, but I do not insist on that, it just feels more understandable that way

**Review by vitek-rostislav — CHANGES_REQUESTED** [review]
Hi, I have added a few suggestions, and also, check the failing tests, please.

**Review by vitek-rostislav — APPROVED** [review]

**Review by pk16011990 — APPROVED** [review]

**Pull request merged**